docs/concepts.rst

```diff
@@ -263,6 +263,31 @@ and equivalent to:
 
     op1.set_downstream([op2, op3])
 
+Building linear DAGs with unary + operator
+------------------------------------------
+Added in Airflow 10.1.3 (?)
+
+In simple, linear DAG case, operator relationships can be set with the unary + operator. The first time the
++ operator is used in DAG it adds no upstream dependency for the task, every subsequent time it adds upstream
+dependency to the task that was most recently added with the + operator. Unlike with bitshift operators
+the operators have to have the same dag already assigned (dag assignment does not carry through.
+
+.. code:: python
+
+    +op1
+    +op2
+    +op3
```

Comment:
I'm not   sure about this syntax -- it feels a bit too magic and hard to work out what it  _means_/is doing.

(I can see the usefulness of the idea though)

Reply:
@ashb @feluelle: I agree it's not really obvious what (+x) does. It's really mostly convenience for some examples. 

It's really convention that we might get used to eventually. 

Unfortunately we are limited with only 3 operators: +, -, ~ : those are the only unary operators we have in Python. Quite obviously '-' is not a good idea, but maybe '~' would look better ?
```
~op1
~op2
~op3
```
Alternatively we could simply use a standard method to achieve similar functionality:

```
op1.upstream_previous()
op2.upstream_previous()
op3.upstream_previous()
```

That would also work and might be less "magical" (and equally useful). 

What do you think?

Reply:
I'm not too confident about pushing this as a preferred way of chaining the operators. My preference would be using the `>>` which will translate into an arrow in the GUI. My main objection is that it feels a bit magic, and by the words of Python Zen:
```
MacBook-Pro-van-Fokko:incubator-airflow fokkodriesprong$ python3
Python 3.6.6 (v3.6.6:4cf1f54eb7, Jun 26 2018, 19:50:54) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import this
The Zen of Python, by Tim Peters

Beautiful is better than ugly.
Explicit is better than implicit.
Simple is better than complex.
Complex is better than complicated.
Flat is better than nested.
Sparse is better than dense.
Readability counts.
Special cases aren't special enough to break the rules.
Although practicality beats purity.
Errors should never pass silently.
Unless explicitly silenced.
In the face of ambiguity, refuse the temptation to guess.
There should be one-- and preferably only one --obvious way to do it.
Although that way may not be obvious at first unless you're Dutch.
Now is better than never.
Although never is often better than *right* now.
If the implementation is hard to explain, it's a bad idea.
If the implementation is easy to explain, it may be a good idea.
Namespaces are one honking great idea -- let's do more of those!
```
I think explicit is better than implicit. I'm also not a big fan of the `.upstream_previous()`, my main preference would be the `>>` and `<<` operator (as in all the examples).

Reply:
I would not want to make it a preferred way at all. It's more convenience for development and testing time. 

But I just realised by reading your "preferred" above @Fokko , that indeed putting it in examples is not a good idea as people might take it literally as "example to follow". That would be misleading and the pain point to solve is not as big and not worth it likely.

I will close this one and we will fix soon the next_dep to use standard >>.